applications/RANSModellingApplication/custom_strategies/generic_residualbased_simple_steady_scalar_scheme.h

```diff
@@ -0,0 +1,377 @@
+
+    using GeometryType = Element::GeometryType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    GenericResidualBasedSimpleSteadyScalarScheme(const double RelaxationFactor)
+        : mRelaxationFactor(RelaxationFactor)
+    {
+        KRATOS_INFO("GenericResidualBasedSimpleSteadyScalarScheme")
+            << " Using residual based simple steady scheme with relaxation "
+               "factor = "
+            << std::scientific << mRelaxationFactor << "\n";
+    }
+
+    ~GenericResidualBasedSimpleSteadyScalarScheme() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void InitializeSolutionStep(ModelPart &r_model_part,
+                                        TSystemMatrixType &A,
+                                        TSystemVectorType &Dx,
+                                        TSystemVectorType &b) override
+    {
+        BaseType::InitializeSolutionStep(r_model_part,A,Dx,b);
+        // if (TSparseSpace::Size(mPreviousB) != TSparseSpace::Size(b)) {
+        //     TSparseSpace::Resize(mPreviousB, TSparseSpace::Size(b));
+        // }
+        // TSparseSpace::SetToZero(mPreviousB);
+        mIterationCounter = 0;
+    }
+
+    void Update(ModelPart& rModelPart,
+                DofsArrayType& rDofSet,
+                TSystemMatrixType& rA,
+                TSystemVectorType& rDx,
+                TSystemVectorType& rb) override
+    {
+        KRATOS_TRY;
+
+        mpDofUpdater->UpdateDofs(rDofSet, rDx, mRelaxationFactor);
+
+        KRATOS_CATCH("");
+    }
+
+    void Clear() override
+    {
+        this->mpDofUpdater->Clear();
+    }
+
+    void CalculateSystemContributions(Element::Pointer rCurrentElement,
+                                      LocalSystemMatrixType& LHS_Contribution,
+                                      LocalSystemVectorType& RHS_Contribution,
+                                      Element::EquationIdVectorType& EquationId,
+                                      ProcessInfo& CurrentProcessInfo) override
+    {
+        KRATOS_TRY;
+
+        const GeometryType& r_geometry = rCurrentElement->GetGeometry();
+        const double local_delta_time = this->GetGeometryLocalTimeStep(r_geometry);
+
+        rCurrentElement->SetValue(DELTA_TIME, local_delta_time);
+
+        rCurrentElement->InitializeNonLinearIteration(CurrentProcessInfo);
+        rCurrentElement->CalculateLocalSystem(
+            LHS_Contribution, RHS_Contribution, CurrentProcessInfo);
+
+        Matrix SteadyLHS;
+        rCurrentElement->CalculateLocalVelocityContribution(
+            SteadyLHS, RHS_Contribution, CurrentProcessInfo);
+        rCurrentElement->EquationIdVector(EquationId, CurrentProcessInfo);
+
+        if (SteadyLHS.size1() != 0)
+            noalias(LHS_Contribution) += SteadyLHS;
+
+        // AddRelaxation(rCurrentElement->GetGeometry(), local_delta_time,
+        //               LHS_Contribution, RHS_Contribution, CurrentProcessInfo);
```

Comment:
There's a few instances of old code commented out in this file, please remove it.